src/DeployRunner.php

```diff
@@ -210,12 +210,25 @@ private function configureStageServer(Stage $stage, Server $server, Configuratio
             );
             $host->set($key, $value);
         }
+        foreach ($config->getVariables('all:'.$stage->getName()) as $key => $value) {
```

Comment:
Please apply PSR 12 formatting :)

Reply:
Also for the other parts of this patch

Reply:
I fixed a bug for missing variable and combined the four loops into one loop that goes through the four stages.

Reply:
Good suggestion! One more question, does it make sense to have stage specific variables for the build step? How does build for production for example? As far as I know you can only build one variant.

Reply:
When you call `$configuration->setVariable('build_number', 5, "build:production")`. This will only be available in production builds.

Reply:
Okay just curious, how do you build for production? Do you set an env variable? Because it's not possible to pass a stage to the `build` command.

Reply:
@tdgroot We haven't started using this. Even with the change we could not get the specific problem fixed. As we could not change or add a task after the symlink task. We had to use another solution.